blog/content/post/2025-04-10-CROSSCON-its-Hypervisor-and-Zarhus.md

```diff
@@ -0,0 +1,481 @@
+[    2.961693] Kernel Offset: 0x80000 from 0xffffffc080000000
+[    2.967250] PHYS_OFFSET: 0x0
+[    2.970162] CPU features: 0x0,80000200,3c020000,0000421b
+[    2.975543] Memory Limit: none
+[    2.978634] ---[ end Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0) ]---
+```
+
+This was an oversight on my part - the rootfs was definitely there on the second
+partition, but I wasn't telling the kernel properly on where it was. Because of
+this, the kernel couldn't find it, and it panicked.
+
+This was fixed in the device tree file, by specifying the correct partition.
+All it took was to add `root=/dev/mmcblk1p2 rw rootwait` to `bootargs` - this
+way the kernel knows where the rootfs is, and will wait until it is mounted.
+
+I thought that was the end of it for the rootfs mounting, but quickly realized
+that it was never going to be this easy - turns out that this location must
+also be specified on the rootfs itself, in `/etc/fstab`. And by default our
+Yocto environment generates this `fstab` file in such a way, that works when
+using it normally, but causes issues with the Hypervisor setup.
+
+All that needs to be done is to change `/dev/mmcblk0p1` to `/dev/mmcblk1p1`
+in the last line:
+
+```bash
+user in ~ λ cat /mnt/etc/fstab
+# stock fstab - you probably want to override this with a machine specific one
+
+/dev/root            /                    auto       defaults              1  1
+proc                 /proc                proc       defaults              0  0
+devpts               /dev/pts             devpts     mode=0620,ptmxmode=0666,gid=5      0  0
+tmpfs                /run                 tmpfs      mode=0755,nodev,nosuid,strictatime 0  0
+tmpfs                /var/volatile        tmpfs      defaults              0  0
+
+# uncomment this if your device has a SD/MMC/Transflash slot
+#/dev/mmcblk0p1       /media/card          auto       defaults,sync,noauto  0  0
+
+/dev/mmcblk1p1  /boot   vfat    defaults    0   0
+user in ~ λ
+```
+
+### Issues logging in
+
+I really thought it was the end of weird fixes by then, but there was one final
+one. I was running into problems when the kernel was booting without any errors,
+but suddenly freezing at some point. Initially I expected it to be a login
+issue, so I was looking at the login service and other related things.
+
+This turned out to not be the cause after all - we
+[got info](https://github.com/crosscon/CROSSCON-Hypervisor-and-TEE-Isolation-Demos/issues/8#issuecomment-2702293550)
+that repeatedly printing a message in a specific abort handler within the
+Hypervisor fixes the issue of freezing and not being able to log in.
+
+This is a Hypervisor related issue, and this is just a temporary workaround -
+but it allows us to use the setup with `rootfs` and Zarhus.
+
+### Summary
+
+The result of all this debugging is a
+[ready-to-follow](https://docs.zarhus.com/guides/rpi4-crosscon-hypervisor/)
+guide on how this can be achieved.
```

Comment:
How "what" can be achieved? After readin to this point I have not idea what "this" is ;)